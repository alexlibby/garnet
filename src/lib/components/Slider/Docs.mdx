import { Canvas, Meta, Story } from "@storybook/addon-docs";
import { ArgTypes } from "@storybook/blocks";
import * as SliderStories from "./Slider.stories";

<Meta of={SliderStories} />

# Slider

Slider is the primary component. It has three possible states.

- [Default](#text)
- [Disabled](#disabled)
- [Step](#step)

## Default

This is the default state for the Slider component:

<Canvas of={SliderStories.Default} />

## Step

This is the Slider component using a step value of 25, and with ticks set to true:

<Canvas of={SliderStories.Step} />

## Disabled

This is the Slider component when disabled:

<Canvas of={SliderStories.Disabled} />

## Properties of component

Below is a list of arguments available for this component:

<ArgTypes of={SliderStories} />
