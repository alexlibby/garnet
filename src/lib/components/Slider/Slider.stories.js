import Slider from "./Slider.svelte";
import { BADGE } from "storybook-addon-badges";

const BADGES = {
  ...BADGE,
  ALPHA: "Alpha",
};

export default {
  title: "Garnet UI Library/Basic Components/Slider",
  component: Slider,
  label: "Example Slider",
  argTypes: {
    val: 1,
    min: 0,
    max: 100,
    step: 10,
    ticks: false,
  },
};

export const Default = {
  args: {
    label: "Text:",
  },
};

Default.parameters = {
  badges: [BADGES.ALPHA],
  docs: { source: { type: "code" } },
};

export const Step = {
  args: {
    step: 25,
    ticks: true,
  },
};

Step.parameters = {
  badges: [BADGES.ALPHA],
  docs: { source: { type: "code" } },
};

export const Disabled = {
  args: {
    disabled: true,
  },
};

Disabled.parameters = {
  badges: [BADGES.ALPHA],
  docs: { source: { type: "code" } },
};
