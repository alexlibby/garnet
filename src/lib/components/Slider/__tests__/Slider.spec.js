import { describe, test, expect } from "vitest";
import { axe } from "vitest-axe";
import * as AxeMatchers from "vitest-axe/matchers";
import { render, screen, fireEvent } from "@testing-library/svelte";
import Slider from "../../Slider/Slider.svelte";

const SliderHTML = `<div class="garnet-slider s-owSWZ-X6GUc2"><div class="range-slider s-owSWZ-X6GUc2"><label for="sliderChoice" class="s-owSWZ-X6GUc2">Text:</label> <input type="range" id="sliderChoice" min="0" max="100" step="1" name="sliderChoice" class="s-owSWZ-X6GUc2"> </div> <div class="selectedValue s-owSWZ-X6GUc2">Value: 50</div></div>`;
expect.extend(AxeMatchers);

describe("Tests for Slider", () => {
  const props = {
    id: "Slider",
  };

  it("should render properly", () => {
    const result = render(Slider, { props });
    expect(() => result).not.toThrow();
  });

  it("should show a new value when slider handle moved", async () => {
    render(Slider, props);
    const slider = screen.getByRole("slider");
    fireEvent.change(slider, { target: { value: "23" } });
    expect(slider.value).toBe("23");
  });

  it("get a snapshot of component", () => {
    // Create a new container for the test and mount component
    const host = document.createElement("div");
    document.body.appendChild(host);
    const instance = new Slider({ target: host, props: props });

    // check that all rendered OK
    expect(instance).toBeTruthy();

    // Take snapshot
    expect(host.innerHTML).toMatchSnapshot();
  });

  test("should demonstrate no issues with accessibility", async () => {
    const render = () => SliderHTML;
    const result = render();

    const results = await axe(result, {
      rules: {
        region: { enabled: false },
      },
    });

    expect(results).toHaveNoViolations();
  });
});
