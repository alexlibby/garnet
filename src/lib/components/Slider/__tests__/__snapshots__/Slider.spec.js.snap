// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for Slider > get a snapshot of component 1`] = `"<div class="garnet-slider s-owSWZ-X6GUc2"><div class="range-slider s-owSWZ-X6GUc2"><label for="Slider" class="s-owSWZ-X6GUc2">Please select a value:</label> <input type="range" id="Slider" min="0" max="100" step="1" name="Slider" class="s-owSWZ-X6GUc2"> </div> <div class="selectedValue s-owSWZ-X6GUc2">Value: 50</div></div><!--<Slider>-->"`;
